awscli/errorhandler.py

```diff
@@ -11,68 +11,110 @@
-        super(BaseOperationError, self).__init__(msg)
-        self.error_code = error_code
-        self.error_message = error_message
-        self.error_type = error_type
-        self.operation_name = operation_name
-        self.http_status_code = http_status_code
-
-
-class ClientError(BaseOperationError):
-    pass
-
-
-class ServerError(BaseOperationError):
-    pass
-
-
-class ErrorHandler(object):
-    """
-    This class is responsible for handling any HTTP errors that occur
-    when a service operation is called.  It is registered for the
-    ``after-call`` event and will have the opportunity to inspect
-    all operation calls.  If the HTTP response contains an error
-    ``status_code`` an appropriate error message will be printed and
-    the handler will short-circuit all further processing by exiting
-    with an appropriate error code.
-    """
-
-    def __call__(self, http_response, parsed, model, **kwargs):
-        LOG.debug('HTTP Response Code: %d', http_response.status_code)
-        error_type = None
-        error_class = None
-        if http_response.status_code >= 500:
-            error_type = 'server'
-            error_class = ServerError
-        elif http_response.status_code >= 400 or http_response.status_code == 301:
-            error_type = 'client'
-            error_class = ClientError
-        if error_class is not None:
-            code, message = self._get_error_code_and_message(parsed)
-            raise error_class(
-                error_code=code, error_message=message,
-                error_type=error_type, operation_name=model.name,
-                http_status_code=http_response.status_code)
-
-    def _get_error_code_and_message(self, response):
-        code = 'Unknown'
-        message = 'Unknown'
-        if 'Error' in response:
-            error = response['Error']
-            return error.get('Code', code), error.get('Message', message)
-        return (code, message)
+class BaseExceptionHandler:
+    def handle_exception(self, exception, stdout, stderr):
+        raise NotImplementedError('handle_exception')
+
+
+class FilteredExceptionHandler(BaseExceptionHandler):
+    EXCEPTIONS_TO_HANDLE = ()
+    MESSAGE = '%s'
+
+    def handle_exception(self, exception, stdout, stderr):
+        if issubclass(exception.__class__, self.EXCEPTIONS_TO_HANDLE):
+            return_val = self._do_handle_exception(exception, stdout, stderr)
+            if return_val is not None:
+                return return_val
+
+    def _do_handle_exception(self, exception, stdout, stderr):
+        stderr.write("\n")
+        stderr.write(self.MESSAGE % exception)
+        stderr.write("\n")
+        return self.RC
+
+
+class ParamValidationErrorsHandler(FilteredExceptionHandler):
+    EXCEPTIONS_TO_HANDLE = (
+        ParamError, ParamSyntaxError, ArgParseException,
+        ParamValidationError, BotocoreParamValidationError,
+        UnknownArgumentError
```

Comment:
Should we be trying to handle `UnknownArgumentError` here? It seems like it gets caught by `UnknownArgumentErrorHandler` already.